DuckDB.NET.Data/Internal/Writer/EnumVectorDataWriter.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using DuckDB.NET.Native;
+
+namespace DuckDB.NET.Data.Internal.Writer;
+
+internal sealed unsafe class EnumVectorDataWriter : VectorDataWriterBase
+{
+    private readonly DuckDBType enumType;
+
+    private readonly uint enumDictionarySize;
+
+    private readonly Dictionary<string, uint> enumValues;
+
+    public EnumVectorDataWriter(IntPtr vector, void* vectorData, DuckDBLogicalType logicalType, DuckDBType columnType) : base(vector, vectorData, columnType)
+    {
+        enumType = NativeMethods.LogicalType.DuckDBEnumInternalType(logicalType);
+        enumDictionarySize = NativeMethods.LogicalType.DuckDBEnumDictionarySize(logicalType);
+
+        uint maxEnumDictionarySize = enumType switch
+        {
+            DuckDBType.UnsignedTinyInt => byte.MaxValue,
+            DuckDBType.UnsignedSmallInt => ushort.MaxValue,
+            DuckDBType.UnsignedInteger => uint.MaxValue,
+            _ => throw new NotSupportedException($"The internal enum type must be utinyint, usmallint, or uinteger."),
+        };
+        if (enumDictionarySize > maxEnumDictionarySize)
+        {
+            // This exception should only be thrown if the DuckDB library has a bug.
+            throw new InvalidOperationException($"The internal enum type is \"{enumType}\" but the enum dictionary size is greater than {maxEnumDictionarySize}.");
```

Comment:
Hi @Giorgi,

i checked the refactoring what you have made. 👍 I have seen you removed this InvalidOperationException with your refactoring, was that planned or made by mistake?

Reply:
I removed it because it should never happen. If DuckDB has such internal issues we will have bigger problems 🙂

Reply:
Ok that is true :) you can then also remove the maxEnumDictionarySize variable in the constructor, that is then not needed anymore :).